solutions/cf-616D.mdx

```diff
@@ -2,20 +2,67 @@
 id: cf-616D
 source: CF
 title: Longest k-Good Segment
-author: Sofia Yang
+author: Sofia Yang, Mohammad Nour Massri
 ---
 
 [Official Editorial](https://codeforces.com/blog/entry/22712)
+## Implementation
 
+**Time Complexity:** $\mathcal{O}(N)$
 <LanguageSection>
+<CPPSection>
+```cpp 
 
-<JavaSection>
+#include <bits/stdc++.h>
+using namespace std;
+int n,k;
+int a[500009];
+int frequency[1000009];//frequency of each number in the window
+int maxSegment,leftMaxSegment,rightMaxSegment;//max size of the segment and its ends
 
-## Solution
+int main(){
+    scanf("%d%d",&n,&k);//large input data use scanf/printf instead of cin/cout
+    for(int i=0;i<n;i++)
+        scanf("%d",a+i);
 
-**Time Complexity:** $\mathcal{O}(N)$
+    int r=-1;//the right end of the current window
+    int cntDifVal = 0;//the number of different values in the window
+
+    //if frequency[val] == 0 then adding val to the window will result in
+    //adding a new different element which increases the number of different
+    //values by 1
+
+    for(int l=0;l<n;l++){//l is the left end of the window
+
+        //we can expand the window [l,r] to [l,r+1] if these 2 conditions hold:
+        //1. r < n-1 making sure we don't go beyond the array's boundaries
+        //2. adding the value a[r+1] to the window does not increase the
+        //number of different values in the window to be larger than k
 
-## Java Implementation
+        while(r<n-1 && cntDifVal+(frequency[a[r+1]]==0) <= k){
+            cntDifVal += frequency[a[r+1]]==0;//update the number of different values
+            frequency[a[r+1]]++;//update the frequency array
+            r++;//update the right end of the current window
+        }
+
+		//update the answer if we find a larger segment
+        if(r-l+1 > maxSegment){
+            maxSegment = r-l+1;
+            leftMaxSegment = l;
+            rightMaxSegment = r;
+        }
+
+        //here we discard the first element a[l] making the current segment [l+1,r]
+        frequency[a[l]]--;//update the frequency
+        cntDifVal -= frequency[a[l]] == 0;//update the number of different values
+    }
+    // we add one because the answer should be 1-indexed
```

Comment:
Having either a space or nothing after ```//``` are both ok, but try to keep the comment style consistent.